### PR TITLE
Patch aws_directory_service_directory Read to avoid a panic

### PR DESCRIPTION
**Pull request by t0yv0**
Fixes https://github.com/pulumi/pulumi-aws/issues/1152

Since it is difficult to repro at integration level, adding a unit test instead.

### CODE REVIEW

File: patches/0055-Fix-panic-in-aws_directory_service_directory-Read.patch [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
++// Copyright (c) HashiCorp, Inc.
++// SPDX-License-Identifier: MPL-2.0
++
++package ds
++
++import (
++	"testing"
++
++	"github.com/aws/aws-sdk-go/service/directoryservice"
++	//"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
```
Comment (corymhall) [inline patches/0055-Fix-panic-in-aws_directory_service_directory-Read.patch:61]: Forgot a comment

Comment (t0yv0) [reply patches/0055-Fix-panic-in-aws_directory_service_directory-Read.patch]: Fixed now.

File: patches/0055-Fix-panic-in-aws_directory_service_directory-Read.patch [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
++
++package ds
++
++import (
++	"testing"
++
++	"github.com/aws/aws-sdk-go/service/directoryservice"
++)
++
++func TestResourceDirectoryReadDoesNotPanicOnMissingVpcSettings(t *testing.T) {
```
Comment (VenelinMartinov) [inline patches/0055-Fix-panic-in-aws_directory_service_directory-Read.patch:63]: Do we/should we run upstream tests in our CI?

Comment (t0yv0) [reply patches/0055-Fix-panic-in-aws_directory_service_directory-Read.patch:63]: I'll look into it. We should but possibly don't.

Comment (VenelinMartinov) [reply patches/0055-Fix-panic-in-aws_directory_service_directory-Read.patch:63]: I'm pretty certain we don't for GCP, so likely we don't for AWS - I imagine some of our patches won't pass tests, so likely to be some effort to make them work.

Just wanted to make sure you are not surprised by this.

Comment (t0yv0) [reply patches/0055-Fix-panic-in-aws_directory_service_directory-Read.patch:63]: It think it's an effort worth taking for our team.

Comment (t0yv0) [reply patches/0055-Fix-panic-in-aws_directory_service_directory-Read.patch:63]: https://github.com/pulumi/pulumi-aws/issues/3022 yes it is still out there. I'll reprioritize this

**Pull request merged**